client/client_test.go

```diff
@@ -568,3 +926,115 @@ func TestPostAndParse(t *testing.T) {
 		})
 	}
 }
+
+func TestAddChain(t *testing.T) {
+	tests := []struct {
+		name        string
+		url         string
+		pemChain    []string
+		statusCode  int
+		rspBody     []byte
+		wantErrType reflect.Type
+		wantSCT     *ct.AddChainResponse
+	}{
+		{
+			name:        "get error",
+			url:         "not-a-real-url",
+			pemChain:    []string{leafPEM, intermediatePEM, rootPEM},
+			wantErrType: reflect.TypeOf(&PostError{}),
+		},
+		{
+			name:        "HTTP status error",
+			pemChain:    []string{leafPEM, intermediatePEM, rootPEM},
+			statusCode:  http.StatusNotFound,
+			wantErrType: reflect.TypeOf(&HTTPStatusError{}),
+		},
+		{
+			name:        "JSON Parse Error",
+			pemChain:    []string{leafPEM, intermediatePEM, rootPEM},
+			statusCode:  http.StatusOK,
+			rspBody:     []byte("not-valid-json"),
+			wantErrType: reflect.TypeOf(&JSONParseError{}),
+		},
+		{
+			name:        "SCT missing Log ID",
+			pemChain:    []string{leafPEM, intermediatePEM, rootPEM},
+			statusCode:  http.StatusOK,
+			rspBody:     []byte(sctMissingID),
+			wantErrType: reflect.TypeOf(&ResponseToStructError{}),
+		},
+		{
+			name:       "SCT missing Timestamp",
+			pemChain:   []string{leafPEM, intermediatePEM, rootPEM},
+			statusCode: http.StatusOK,
+			rspBody:    []byte(sctMissingTimestamp),
+			wantSCT: &ct.AddChainResponse{
+				ID:        mustB64Decode("CEEUmABxUywWGQRgvPxH/cJlOvopLHKzf/hjrinMyfA="),
+				Timestamp: 0,
+				Signature: mustB64Decode("BAMARjBEAiAJAPO7EKykH4eOQ81kTzKCb4IEWzcxTBdbdRCHLFPLFAIgBEoGXDUtcIaF3M5HWI+MxwkCQbvqR9TSGUHDCZoOr3Q="),
+			},
+		},
+		{
+			name:        "SCT missing Signature",
+			pemChain:    []string{leafPEM, intermediatePEM, rootPEM},
+			statusCode:  http.StatusOK,
+			rspBody:     []byte(sctMissingSig),
+			wantErrType: reflect.TypeOf(&ResponseToStructError{}),
+		},
+		{
+			name:       "no error",
+			pemChain:   []string{leafPEM, intermediatePEM, rootPEM},
```

Comment:
pemChain is the same in every test.  Do you plan to add cases that have a different pemChain?
If not, remove that from the test table, and parse it to []*x509.Certificate up-front, and then do away with the handling of CertificateFromPEM errors in the t.Run()